AbletonMCP_Remote_Script/__init__.py

```
@@ -561,6 +736,314 @@ def _set_tempo(self, tempo):
+                "total_chains": len(device.chains)
+            }
+            return result
+        except Exception as e:
+            self.log_message("Error creating rack chain: " + str(e))
+            raise
+    
+    def _get_device_parameters(self, track_index, device_index, chain_index=-1, rack_device_index=-1):
+        """Get all parameters of a device"""
+        try:
+            if track_index < 0 or track_index >= len(self._song.tracks):
+                raise IndexError("Track index out of range")
```

Ruff (0.14.6)

Abstract `raise` to an inner function

(TRY301)